ruby/ext/google/protobuf_c/message.c

```diff
@@ -139,7 +139,14 @@ int Message_initialize_kwarg(VALUE key, VALUE val, VALUE _self) {
              "Unknown field name in initialization map entry.");
   }
 
-  if (upb_fielddef_label(f) == UPB_LABEL_REPEATED) {
+  if (is_map_field(f)) {
+    if (TYPE(val) != T_HASH) {
+      rb_raise(rb_eArgError,
+               "Expected hashmap as initializer value for map field.");
```

Comment:
Probably should refer to this as a Hash object (not hashmap).

Seems like we should also support initialization with a Google::Protobuf::Map object?

Though maybe it's too confusing if both Hash and Map are supported as initializers, but with different semantics (one copies, one references)?